tests/import_test.go

```diff
@@ -342,16 +344,17 @@ var _ = Describe("[Istio] Namespace sidecar injection", Serial, func() {
 		}, timeout, pollingInterval).Should(BeTrue())
 
 		By("Verify HTTP request error in importer log")
-		Eventually(func() bool {
-			log, _ := f.RunKubectlCommand("logs", importer.Name, "-n", importer.Namespace)
-			if strings.Contains(log, "HTTP request errored") {
-				return true
-			}
-			if strings.Contains(log, "502 Bad Gateway") {
-				return true
-			}
-			return false
-		}, time.Minute, pollingInterval).Should(BeTrue())
+		Eventually(func() string {
+			// Cannot use GetLogs API because it does not return the error (i only says 'the server rejected our request for an unknown reason')
```

Comment:
If you annotate the DV in this test with `AnnDeleteAfterCompletion` you should be able to get rid of the expectation for `502 Bad Gateway` (I assume it's for the case where importer gets deleted quickly?)

Reply:
I don't really know the logic behind this assertions, I simply refactored the test :sweat_smile:. I'll try that.

Reply:
Actually, I got it wrong, this 502 code appears regularly in importer log text.
In that case, I am not sure why we can't use GetLogs?

Reply:
When I use GetLogs, I get a generic error message saying `the server rejected our request for an unknown reason (get pods cdi-deployment-b5f4f866-rbslq)`.

What I suspect is that this test Eventually is not asserting on the pod's log, but rather on the error message produced by the `kubectl` binary.

I tried to replicate it in my machine but I don't know how to run this test (seems like I need `KUBEVIRT_DEPLOY_ISTIO` but just setting the environment variable is not enough).

Reply:
Thanks! I'll work on fixing this then.